tests/core/test_dialogue.py

```diff
@@ -69,3 +113,59 @@ def test_utterances(dialogue_history_1):
         dialogue_history_1.utterances[4].participant
         == DialogueParticipant.AGENT
     )
+
+
+def test_to_dict_d1(dialogue_history_1: Dialogue) -> None:
+    """Tests dialogue export to dictionary.
+
+    Args:
+        dialogue_history_1: Test Dialogue object 1.
+        dialogue_history_2: Test Dialogue object 2.
+    """
+    dialogue_dict_1 = dialogue_history_1.to_dict()
+
+    assert dialogue_dict_1.get("agent") == "agent-001"
+    assert dialogue_dict_1.get("user") == "USR01"
+    assert dialogue_dict_1.get("metadata") is None
+    assert len(dialogue_dict_1.get("conversation")) == 5
+    utterance_1 = dialogue_dict_1.get("conversation")[0]
+    assert utterance_1["utterance"] == "Hello"
+    assert utterance_1.get("slot_values") is None
+
+
+def test_to_dict_d2(dialogue_history_2: Dialogue) -> None:
+    """Tests dialogue export to dictionary.
+
+    Dialogue has annotations and metadata.
+
+    Args:
+        dialogue_history_2: Test Dialogue object 2.
+    """
+    dialogue_dict_2 = dialogue_history_2.to_dict()
+
+    assert dialogue_dict_2.get("agent") == "agent-002"
+    assert dialogue_dict_2.get("user") == "USR02"
+    assert dialogue_dict_2.get("metadata") == {
+        "description": "Dialogue fixture for testing"
+    }
+    assert len(dialogue_dict_2.get("conversation")) == 3
+    last_utterance = dialogue_dict_2.get("conversation")[-1]
+    assert last_utterance["intent"] == "ELICIT"
+    assert last_utterance["slot_values"] == [["COLOR", "color"]]
```

Comment:
Nit: you could also check the intents of the first 2 utterances and that those don't have annotations.

Reply:
Done.